### PR TITLE
'Duplicate an order' button

### PR DESCRIPTION
**Pull request by vladimir-8**
Added a button: `Duplicate an order`: opens "Create a new order" page while coping/pasting data from the current order (no.1 in https://github.com/coopcycle/coopcycle-web/issues/4331 )

Screenshot 2024-07-04 at 12 28 02

### CODE REVIEW

File: templates/delivery/form.html.twig [hunk 1]
```diff
@@ -116,11 +116,18 @@
...
-    <div class="col-md-5">
-      <button id="delivery-submit" type="submit" class="btn btn-block btn-lg btn-primary h-100">
+    <div class="col-md-5 delivery__form__buttons">
+      <button id="delivery-submit" type="submit" class="btn btn-block btn-lg btn-primary">
         <i class="fa fa-spinner fa-spin hidden" id="loader"></i>
         {% trans %}basics.save{% endtrans %}
       </button>
+      {% if is_granted('ROLE_ADMIN') and not is_new and delivery.order is not empty %}
+        <a href="{{ path('admin_store_delivery_new',
+          { id: delivery.store.id, frmrdr: delivery.order|hashid(16) }
```
Comment (r0xsh) [inline templates/delivery/form.html.twig:126]: `delivery.store` is only set for "lastmile" deliveries. This fields is empty for foodtech

Comment (r0xsh) [reply templates/delivery/form.html.twig:126]: Which lead to this error for foodtech deliveries
image

Comment (vladimir-8) [reply templates/delivery/form.html.twig:126]: Fixed in 5ef7b02f81b01894d4031360b8d65a0561da98d7

**Comment by HugoLCR** [comment]
I have a problem with the added function:
the date and time are not respected -> my duplicate delivery switches to the next working day at the time of duplication

**Comment by Paul-Eraman-CoopCycle** [comment]
Hi hugo, could you be more specific? Like, this order was supposed to be X time / day but instead was Y time / day

Or record a gif using licecap?

**Comment by HugoLCR** [comment]
I just need a cloned order, when I duplicate. Not a new order, on the next date.

**Comment by Paul-Eraman-CoopCycle** [comment]
I think the way it is working is the expected behavior. You can modify the day once the order is created no?

Remember, it has to work for multiple different types of situations at once

what types of orders are you copying?

**Comment by HugoLCR** [comment]
I need to duplicate order from the B2B form 3-4x per week, for urgent customer needs, or poorly estimated volumes (2 services).
This meets urgent needs.

**Comment by Paul-Eraman-CoopCycle** [comment]
That can be done by just adding another dropoff at the end with the original address

**Comment by Paul-Eraman-CoopCycle** [comment]
"I need to duplicate order from the B2B form 3-4x per week, for urgent customer needs, or poorly estimated volumes (2 services).
This meets urgent needs."

I dont understand, can we do a video call next week?

**Comment by HugoLCR** [comment]
I'll detail it for you tomorrow

**Comment by HugoLCR** [comment]
https://github.com/user-attachments/assets/5afe4a2c-5b3b-45aa-aa1c-331f6f94bb40

I actually think I didn't make myself understood well.

I just need a duplicate with more :

- a same date
- a same hour
- a same price

**Pull request merged**